### PR TITLE
 refactor(api): user consent

### PR DESCRIPTION
**Pull request by madhavilosetty-intel**
## PR Checklist

- [x] Unit Tests have been added for new changes
- [x] API tests have been updated if applicable
- [x] All commented code has been removed
- [ ] If you've added a dependency, you've ensured license is compatible with Apache 2.0 and clearly outlined the added dependency.

## What are you changing?
Updated API's to cancel, request and send user consent code and added unit tests

### CODE REVIEW

File: src/amt/models/amt_models.ts [hunk 1]
```diff
@@ -94,7 +94,7 @@ export interface AMT_EnvironmentDetectionSettingData extends CIM_SettingData<AMT
 export interface AMT_SystemDefencePolicy extends CIM_ManagedElement<AMT_SystemDefencePolicy> {
   PolicyName?: string
   PolicyPrecedence?: number
-  AntiSpoofingSupport?: number
+  antiSpoofingSupport?: number
```
Comment (rsdmike) [inline src/amt/models/amt_models.ts:97]: This should remain uppercase A, additionally we should rename `AMT_SystemDefencePolicy` to `AMT_SystemDefensePolicy` to match the SDK. https://software.intel.com/sites/manageability/AMT_Implementation_and_Reference_Guide/default.htm?turl=HTMLDocuments%2FWS-Management_Class_Reference%2FAMT_SystemDefensePolicy.htm

File: src/amt/models/amt_models.ts [hunk 2]
```diff
@@ -42,7 +42,7 @@ export interface AMT_GeneralSettings extends CIM_SettingData<AMT_GeneralSettings
 }
 
 export interface AMT_EthernetPortSettings extends CIM_SettingData<AMT_EthernetPortSettings> {
-  VLANTag?: number
+  VlanTag?: number
```
Comment (rsdmike) [inline src/amt/models/amt_models.ts:45]: this should remain `VLANTag`

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/amt/models/amt_models.ts (hunk 2 above)
Comment (madhavilosetty-intel) [reply src/amt/models/amt_models.ts]: resolved

File: src/amt/models/amt_models.ts (hunk 1 above)
Comment (madhavilosetty-intel) [reply src/amt/models/amt_models.ts]: done

File: src/amt/ConnectedDevice.ts [hunk 3]
```diff
@@ -1,13 +1,14 @@
 import { CIRASocket } from '../models/models'
-import { CIM } from './cim/CIM'
-import { Methods } from './cim/methods'
+import { Methods as CIM_Methods, CIM } from './cim/index'
 import { HttpHandler } from './HttpHandler'
 import { CIRAChannel, CIRAHandler } from './CIRAHandler'
 import { logger } from '../utils/logger'
-import { AMT } from './AMT'
+import { Methods as AMT_Methods, AMT } from './amt/index'
+import { Methods as IPS_Methods, IPS } from './ips/index'
```
Comment (rsdmike) [inline src/amt/ConnectedDevice.ts:7]: you shouldn't need to use /index here

**Review by rsdmike — CHANGES_REQUESTED** [review]

File: src/amt/ConnectedDevice.ts (hunk 3 above)
Comment (madhavilosetty-intel) [reply src/amt/ConnectedDevice.ts:7]: done

**Review by rsdmike — APPROVED** [review]

**Pull request merged**